Belated sync

NiceneNerd opened the pull request:
Let's bring all this back home

GingerAvalanche committed:
Build off of reworked RSTB gen

GingerAvalanche committed:
Force Aoc lookup first

GingerAvalanche committed:
Deployment rework

GingerAvalanche committed:
Link aoc folder to Cemu

GingerAvalanche committed:
Link *from* aoc for Cemu

GingerAvalanche committed:
Disguise bug fix as code consolidation

GingerAvalanche committed:
Fix file copying on Windows

GingerAvalanche committed:
Fix file copier for BCML migration

GingerAvalanche committed:
Add rules.txt to direct install options

GingerAvalanche committed:
Allow migration to install folders

GingerAvalanche committed:
Bnp conversion diff against empty

When converting submods, just diff against an empty file, seeing as we have no good interface for converting to ukmm's dependency system

GingerAvalanche committed:
Fix symlink-to-copy switching

GingerAvalanche committed:
Update mods.rs

Fix a rare issue where a mod might not be logged for inclusion if it doesn't edit the same language as the mods in the edit manifest

GingerAvalanche committed:
Move symlink functions to util

GingerAvalanche committed:
Reset output prior to settings change

Deletes the output folder if it can be deleted and if the output folder or method has changed
Then resets pending, to log that another deployment is needed

GingerAvalanche committed:
Handle missing DeleteRev

Wish I could just tell mod authors to stop breaking mods, but since DeleteRev is always -1 anyway...

GingerAvalanche committed:
Add location contexts for errors

GingerAvalanche committed:
Bnp conversion diff against empty pt 2

Similar to da97f4d but for SARCs

GingerAvalanche committed:
Enable vertical scrolling in dep window

GingerAvalanche committed:
Fix context building for non-Windows

GingerAvalanche committed:
Add line numbers to task errors

GingerAvalanche committed:
Check base game was found instead

GingerAvalanche committed:
Fix imported Cemu exe location

GingerAvalanche committed:
Support title_list_cache/WUA

GingerAvalanche committed:
Platform-agnostic cemu exe path insertion

GingerAvalanche committed:
Canonicalize dump paths

Also checks to make sure they exist

GingerAvalanche committed:
Clean mods when deleting profiles

When deleting a profile, remove any mods that are not used by any other profile

GingerAvalanche committed:
Update deploy method tooltip

GingerAvalanche committed:
Remap language diffing

For region mismatch of same language, diff internal SARC instead of external one

GingerAvalanche committed:
Sort user language first

Why, oh why, can programming environments never agree on sort ordering? I dealt with this in JavaScript between Chromium and Gecko, too! Ugh...

GingerAvalanche committed:
Update CHANGELOG.md

GingerAvalanche committed:
Move rstb feature flag out

GingerAvalanche committed:
Version bump

GingerAvalanche committed:
Hijack update version URL

GingerAvalanche committed:
Typo

GingerAvalanche committed:
More hijacking

GingerAvalanche committed:
Still more hijacking

GingerAvalanche committed:
Add context for bnp texts error

GingerAvalanche committed:
Remove unnecessary extension

GingerAvalanche committed:
Don't collect directories

Ensures only files are collected for file copying

GingerAvalanche committed:
Best practice PathBuf join

GingerAvalanche committed:
Find title cache wherever it is

GingerAvalanche committed:
Update unreleased changelog

GingerAvalanche committed:
Don't keep a one-use variable

GingerAvalanche committed:
Consistent style

GingerAvalanche committed:
Make optional emulator path obvious

GingerAvalanche committed:
Clarify Deploy Layout setting

GingerAvalanche committed:
Fix bnp crash regression

GingerAvalanche committed:
Fix Windows using Powershell

GingerAvalanche committed:
Changelog

GingerAvalanche committed:
Version bump

GingerAvalanche committed:
Stupid macOS deprecation

GingerAvalanche committed:
Additional context for profile errors

GingerAvalanche committed:
Profile duplication crash fix

GingerAvalanche committed:
Explicit update asset names

GingerAvalanche committed:
Add cmd cli command option for Windows

GingerAvalanche committed:
Copyright update

GingerAvalanche committed:
Initial localization

GingerAvalanche committed:
Make platform mismatch error more obvious

GingerAvalanche committed:
Add fonts to ouch exe size

GingerAvalanche committed:
Add newline support for loc

GingerAvalanche committed:
Minor en loc clarity

GingerAvalanche committed:
Reset settings resets interface loc

GingerAvalanche committed:
Add archive name to metadata prompt

GingerAvalanche committed:
Changelog

GingerAvalanche committed:
Readme special thanks

GingerAvalanche committed:
Appease Clippy

GingerAvalanche committed:
Format special thanks

GingerAvalanche committed:
Format special thanks username links

GingerAvalanche committed:
Dutch loc

GingerAvalanche committed:
Dutch loc formatting

GingerAvalanche committed:
Add Chinese loc

GingerAvalanche committed:
Fix Deploy Method display string

GingerAvalanche committed:
Fix typo

GingerAvalanche committed:
Add french loc

GingerAvalanche committed:
Add category keys to english

GingerAvalanche committed:
Add italian loc

GingerAvalanche committed:
Remove english line

GingerAvalanche committed:
Chinese category keys

GingerAvalanche committed:
+Categories, -English for other langs

GingerAvalanche committed:
Support for category localization

GingerAvalanche committed:
Special thanks names/links

GingerAvalanche committed:
More readme stuff

GingerAvalanche committed:
Better dutch category loc

GingerAvalanche committed:
German categories fix

GingerAvalanche committed:
Update zhongwen loc

GingerAvalanche committed:
Support text deepmerge

GingerAvalanche committed:
Prep for 0.15.3 release

GingerAvalanche committed:
Fix level sensor ordering

GingerAvalanche committed:
Debug switch logger error

GingerAvalanche committed:
Deploy button always visible

GingerAvalanche committed:
Fix Windows open emu again

GingerAvalanche committed:
Validate profiles on app boot

GingerAvalanche committed:
Changelog

GingerAvalanche committed:
Changelog again

GingerAvalanche committed:
Version bump

GingerAvalanche committed:
Force rules.txt and method on cemu import

GingerAvalanche committed:
Make profile validation work across dualboots

GingerAvalanche committed:
Cemu import emulator command fix

GingerAvalanche committed:
Add MainStatic merger

GingerAvalanche committed:
Filter out copies when source is a folder

GingerAvalanche committed:
Insert new merger to routine

GingerAvalanche committed:
Add context to errors

GingerAvalanche committed:
Make NotUseForStats read optional

GingerAvalanche committed:
Merge MainStatic properly maybe

GingerAvalanche committed:
Error propagation

GingerAvalanche committed:
Clarify vector errors

GingerAvalanche committed:
Only search up the number of masters to use

GingerAvalanche committed:
Make diff entry names more unique

GingerAvalanche committed:
Get data goes dlc->update->base

GingerAvalanche committed:
Vector coercion and warnings

GingerAvalanche committed:
Delete old log on each startup

No more 10gb worth of log files

GingerAvalanche committed:
Make hashes more specific

GingerAvalanche committed:
Push MainStatic IDs into objects

GingerAvalanche committed:
Remove extra bphysics size, rstb-rust handles it

GingerAvalanche committed:
Compress fonts

GingerAvalanche committed:
Dump retrieval rework

GingerAvalanche committed:
Aoc sarc file packing fix

GingerAvalanche committed:
Fix PathBuf parsing for Windows->Unix

GingerAvalanche committed:
Log lines for profile validation

GingerAvalanche committed:
Changelog

GingerAvalanche committed:
Skip packing unmodded binary files

GingerAvalanche committed:
Clean up changelog

GingerAvalanche committed:
Appease the IDE

GingerAvalanche committed:
Fix *nix graphicPack fallback path

GingerAvalanche committed:
Appease the IDE

GingerAvalanche committed:
Optimize Cemu importer behavior

GingerAvalanche committed:
Pull error out of function

GingerAvalanche committed:
Remove unnecessary pathing

GingerAvalanche committed:
Clean up code

GingerAvalanche committed:
Refactor profile validator

Fix hashes
Optimize pathfinding
Add mods that exist in mods folder and load order but not mod list

GingerAvalanche committed:
Wayland DPI runtime fix

GingerAvalanche committed:
Set logger file at program open

GingerAvalanche committed:
Add context to unwraps at merge time

GingerAvalanche committed:
Add context to bnp map rebuilding

GingerAvalanche committed:
Set generic top-level error message

GingerAvalanche committed:
Update IT loc

GingerAvalanche committed:
Add cemu settings file dump path support

GingerAvalanche committed:
Make modifications before additions or deletions

Fixes modifications happening to the wrong instances

GingerAvalanche committed:
Use or_else for function calls

GingerAvalanche committed:
Delete deletables from DeleteSet

Should fix errors where things were never deleted from sets

GingerAvalanche committed:
Always full deploy copy and hardlink

GingerAvalanche committed:
Initial AS stuff

Need to solve references to objects
Currently refers to all objects as their own
Needs to use a list of elements and a map of children

GingerAvalanche committed:
Limit sometimes parsing binary as byml

GingerAvalanche committed:
Changelog

GingerAvalanche committed:
Update pending log manager

GingerAvalanche committed:
Save pending log on pending reset

GingerAvalanche committed:
Changelog

GingerAvalanche committed:
Appease the linter

GingerAvalanche committed:
Add JetBrains files to gitignore

GingerAvalanche committed:
Fix wua test

Previous test was trying to obtain a file that didn't exist
And also, it was testing the update version to be 0.9.0
But the update version can never be 0.9.0, that's the base
game version

GingerAvalanche committed:
Filter diff contents from source in DeleteSet

Simple chaining will remove the diff version when
deleting, so we must filter out the ones that already
existed.

Fixes the fact that dab76d7 didn't actually do what it
was supposed to

GingerAvalanche committed:
NamedEnumerate only extends to capacity

Fixes an error where set_len() cannot be used to
set the length of a vector beyond its capacity,
even in unsafe blocks (which makes sense because
that would never ever ever result in desired code)

GingerAvalanche committed:
Fix 379ea86 on Windows

GingerAvalanche committed:
Fix 0b65d7

What was I thinking?

The0x539 committed:
Remove accidental redirect follow in oneclick handler

The0x539 committed:
Install rustls crypto provider in all code paths

The0x539 committed:
Use a more sane Deserialize impl for FilePickerState

GingerAvalanche committed:
Reuse already-read file in case of pending log load failure

GingerAvalanche committed:
Update localization

Move loc to its own crate, allow for replacing strings
across the entire program. Only existing loc has been
migrated, new loc will be added later.
Change missing loc to display English, instead of the keys

k-carbonatedtea committed:
Fix egui assertion failures in debug mode

Validate widget sizes to ensure they are finite and non-negative before
passing to egui. Fixes panics when ui.available_size() returns invalid
values in certain layout contexts.

GingerAvalanche committed:
Add rar support

GingerAvalanche committed:
Replace all localizations with string extension

GingerAvalanche committed:
Remove test output file

GingerAvalanche committed:
Appease compile checker

For some reason, the actual compiler didn't have any issues building with that error, it was only the checker that threw a fit

GingerAvalanche committed:
Changelog

GingerAvalanche committed:
Fix non-ukmm zips finding modded files

GingerAvalanche committed:
Changelog

GingerAvalanche committed:
Update changelog to reflect actual changes

GingerAvalanche committed:
Fix error when pending file doesn't exist

Clonephaze committed:
Intro Update | New User Helper Messages

Intro:
- New text for a more immediately helpful set of instructions
- Set initial width and height to show entire message
- Now stored in markdown format in the localization file for future translation.

Helper Messages
- Empty modlists now display messages intended to be helpful to new users
- If UKMM is not configured yet, displays a message telling users where to go
- If configured but no mods installed, displays a message telling the 3 ways to install new mods

Clonephaze committed:
Add Spanish localization

Clonephaze committed:
Improve German translations and wording

Clonephaze committed:
Update fr.json

Clonephaze committed:
Update ru.json

Clonephaze committed:
Update nl.json

Clonephaze committed:
Update ko.json

Clonephaze committed:
Update ja.json

Clonephaze committed:
Update zh.json

Clonephaze committed:
Created Arabic File

Clonephaze committed:
Created Vietnamese file

Clonephaze committed:
Add Arabic and Vietnamese localizations

Include Arabic (ar.json) and Vietnamese (vi.json) locale files and wire them into localization logic.

GingerAvalanche committed:
Revert changes to human-translated loc

GingerAvalanche committed:
Update Italian loc with new keys

GingerAvalanche committed:
Update Changelog

GingerAvalanche committed:
Update Changelog again

GingerAvalanche committed:
Gracefully handle IPC socket collision

GingerAvalanche committed:
APPEEEEEEAAAAAAAASE THE LINTERRRRRRR

GingerAvalanche commented:
LINTER APPEASED.

I am _so, so sorry_ about all the literally _hundreds_ of emails you suffered as a result of me taking so long to figure out how to use clippy in my IDE

GingerAvalanche committed:
Don't error on unchanged values, just store nothing

GingerAvalanche committed:
Don't stick the Nones from the diffs into the merged

GingerAvalanche committed:
More linter appeasement

GingerAvalanche committed:
Filter out incomplete objects from being converted back

Accounts for when a MainStatic made from the DLC file is applied to a base game MainField/Static.smubin

GingerAvalanche committed:
Exclude possibly-missing param from completeness filter

GingerAvalanche committed:
Exclude THE OTHER possibly-missing param from completeness filter

GingerAvalanche committed:
Enspecify errors

Make errors with multiple contexts display more than just the top-most error, so that people stop claiming they get the same error.

This is similar to the change that generifies the top-most error message, but I like this one more because it actually gives some info

GingerAvalanche committed:
Ugh why do I do this to myself?

There's like 5 commits worth of changes in here. Screw it. Uh...
Appeased the linter
Implemented diff and merge behavior for subtypes in case I ever figure out how to implement it for the overall AnimSeq type
Implemented diff and merge behavior for AnimSeq type as just passing the bas file through to the end
Implemented and hooked the traverser into the initial AnimSeq conversion
Turned Element extensions into a map of extension type: Extension so that we don't try to diff different extension types
Probably some other stuff, I don't know

GingerAvalanche committed:
Appease the linter some more

GingerAvalanche committed:
Implemented diff and merge logic for Extensions enum

GingerAvalanche committed:
Replace AS merger with AnimSeq merger

GingerAvalanche committed:
Appeasing the linter for *one more line*

GingerAvalanche committed:
Convert to using UKError for consistency

GingerAvalanche committed:
Box ContextData in UKError to make it smaller

Linter appeasement!

GingerAvalanche committed:
Unify and clarify AnimSeq errors

Use MissingAampKey for missing aamp keys (novel idea!)
Specify Self type in invalid-parameter-type errors

GingerAvalanche committed:
Make Extensions optional

GingerAvalanche committed:
Clarify AnimSeq Element parse error contexts

GingerAvalanche committed:
Only traverse if there are elements to traverse

GingerAvalanche committed:
Better wording for traversal failure

GingerAvalanche committed:
Increase hash lookups to 256 values

Probably way overkill, but better than failure I guess

GingerAvalanche committed:
Clarify key->index mapping error messages

GingerAvalanche committed:
Insert Extend into Element lists

GingerAvalanche committed:
Remove repo move notice

GingerAvalanche committed:
Revert hijacking in prep for migration back to official repo

GingerAvalanche committed:
Changelog

GingerAvalanche committed:
Version bump

GingerAvalanche committed:
Readme - CMake version

GingerAvalanche committed:
Comment unused crates, thanks Clippy...

GingerAvalanche committed:
Remove (obsolete?) multilingual param from book.toml
